discovery_engine_core/core/src/storage.rs

```diff
@@ -216,11 +212,21 @@ pub mod models {
 
     impl From<ApiDocumentView> for document::Document {
         fn from(view: ApiDocumentView) -> Self {
+            let ApiDocumentView {
+                document_id,
+                news_resource,
+                newscatcher_data,
+                user_reaction,
+                embedding,
+                stack_id,
+            } = view;
```

Comment:
It will trigger a warning when you add fields to `ApiDocumentView` but forgot to add them to `Document`, which you most likely need to do.

Reply:
`ApiDocumentView` is meant to be the view which contains the data returned to the app, i.e. the data in the api, i.e. what is in `api/models/document`.

And `Document` is the type we return from the engine to dart.

Long term I expect:
1. the internals of the engine to use a smaller view, e.g. currently the whole stack algorithm only uses I think 3 or 4 fields of the whole document including embedding and news resource
2. at the boundary to fetch the remaining data the app needs, i.e. the `ApiDocumentView` 
3. in turn `document::Document` will likely be removed at some point

Reply:
i agree with the long term points. i still don't agree with the need for destructuring, but let's keep it since it's going to be changed later anyways.